WordPress/src/main/java/org/wordpress/android/ui/main/WPMainNavigationView.kt

```diff
@@ -137,35 +147,39 @@ class WPMainNavigationView @JvmOverloads constructor(
     override fun onNavigationItemReselected(item: MenuItem) {
         // scroll the active fragment's contents to the top when user re-taps the current item
         val position = getPositionForItemId(item.itemId)
-        if (position != PAGE_NEW_POST) {
+        if (position != getPosition(NEW_POST)) {
             (navAdapter.getFragment(position) as? OnScrollToTopListener)?.onScrollToTop()
         }
     }
 
     private fun getPositionForItemId(@IdRes itemId: Int): Int {
         return when (itemId) {
-            R.id.nav_sites -> PAGE_MY_SITE
-            R.id.nav_reader -> PAGE_READER
-            R.id.nav_write -> PAGE_NEW_POST
-            R.id.nav_me -> PAGE_ME
-            else -> PAGE_NOTIFS
+            R.id.nav_sites -> getPosition(MY_SITE)
+            R.id.nav_reader -> getPosition(READER)
+            R.id.nav_write -> getPosition(NEW_POST)
+            R.id.nav_me -> getPosition(ME)
+            else -> getPosition(NOTIFS)
         }
     }
 
+    private fun getPageForItemId(@IdRes itemId: Int): PageType {
+        return getPageType(getPositionForItemId(itemId))
+    }
+
     @IdRes
     private fun getItemIdForPosition(position: Int): Int {
-        return when (position) {
-            PAGE_MY_SITE -> R.id.nav_sites
-            PAGE_READER -> R.id.nav_reader
-            PAGE_NEW_POST -> R.id.nav_write
-            PAGE_ME -> R.id.nav_me
+        return when (pages().getOrNull(position)) {
```

Comment:
What do you think about pulling this `pages().getOrNull(position)` into a separate method, something like `getPositionOrNull(position)` in the companion object?

Reply:
Makes sense 👍. Changes here 0708529161ebc4f065f0142d206f9a5a8d09bc3b